PFR/ForMathlib/Entropy/RuzsaDist.lean

```diff
@@ -206,10 +212,10 @@ lemma rdist_zero_eq_half_ent [IsFiniteMeasure μ] [IsProbabilityMeasure μ'] :
               measurability
               measurability
     simp [entropy_def, h]
-  simp [rdist_def, entropy_const (0 : G), aux]
+  simp only [rdist_def, aux, entropy_const (0 : G), zero_div, sub_zero]
```

Comment:
Why this change?

Reply:
Ops, I didn't realise that `simp` was used as a finishing tactic.

Reply:
I'm still learning what's the optimal way to clean the code. For now the rule I'm following is "if a `simp` is non-finishing, then substitute it with `simp only [...]`"

Am I wrong?

Reply:
Yes, that's inaccurate.

simp is a "flexible output" tactic, namely it can leave the locations (goal or hypotheses) it touches in wildly different states depending on what's exactly in the simp set. This means that the tactics following it should be "flexible input" tactics, like `simp`, `ring`, `measurability`... This is to ensure that proofs don't break so easily when the simp set is refactored. So simp should be squeezed when in front of "rigid input" tactics, like `refine`, `apply`, `exact`, `rw`

There is however no reason to squeeze it when it's in front of flexible input tactics.

Reply:
I see. Thank you very much for the explanation. I'm going to update this PR accordingly as soon as I find the time.

Reply:
Thank you very much!